src/coreComponents/physicsSolvers/fluidFlow/CompositionalMultiphaseBase.cpp

```diff
@@ -789,8 +832,25 @@ void CompositionalMultiphaseBase::initializeFluidState( MeshLevel & mesh,
       SolidInternalEnergy const & solidInternalEnergyMaterial =
         getConstitutiveModel< SolidInternalEnergy >( subRegion, solidInternalEnergyName );
       solidInternalEnergyMaterial.saveConvergedState();
+    }
 
+    // Step 4.7: if the diffusion and/or dispersion is/are supported, initialize the two models
+    if( m_hasDiffusion )
+    {
+      string const & diffusionName = subRegion.template getReference< string >( viewKeyStruct::diffusionNamesString() );
+      DiffusionBase const & diffusionMaterial = getConstitutiveModel< DiffusionBase >( subRegion, diffusionName );
+      arrayView1d< real64 const > const temperature = subRegion.template getField< fields::flow::temperature >();
+      diffusionMaterial.initializeTemperatureState( temperature );
     }
+    if( m_hasDispersion )
+    {
+      string const & dispersionName = subRegion.template getReference< string >( viewKeyStruct::dispersionNamesString() );
+      DispersionBase const & dispersionMaterial = getConstitutiveModel< DispersionBase >( subRegion, dispersionName );
+      GEOS_UNUSED_VAR( dispersionMaterial );
+      // TODO: compute the total velocity here
```

Comment:
why total velocity? should be phase, no?

Reply:
Yes, if we stick to the SPE11 benchmark description, it must be the phase velocity. @jafranc will handle this part. 

In the current framework, if we use the phase velocity to compute a dispersivity tensor, we have to call `computeWeights` twice in the kernel, which is a little bit annoying. So initially I thought we could use the total velocity instead to avoid this. But @jafranc has a better idea than using the total velocity here and will test it soon to see if it is accurate.

In the meantime, I updated the test that you highlighted to mention phase velocities and not total velocity.